### PR TITLE
URL-encode paths in CurlStorage::MapUTF8

### PR DESCRIPTION
**Pull request by jwise**

### CODE REVIEW

File: src/storage/plugins/CurlStorage.cxx [hunk 1]
```diff
@@ -77,9 +78,18 @@ CurlStorage::MapUTF8(const char *uri_utf8) const noexcept
 	if (StringIsEmpty(uri_utf8))
 		return base;
 
-	// TODO: escape the given URI
+	CurlEasy easy;
+	std::string path_esc;
+
+	for (auto elt: IterableSplitString(uri_utf8, '/')) {
+		char *elt_esc = easy.Escape(elt.data, elt.size);
+		if (path_esc != "")
```
Comment (MaxKellermann) [inline src/storage/plugins/CurlStorage.cxx]: Check `empty()` which is cheaper than string comparison.

File: src/storage/plugins/CurlStorage.cxx [hunk 2]
```diff
@@ -77,9 +78,18 @@ CurlStorage::MapUTF8(const char *uri_utf8) const noexcept
...
+			path_esc = path_esc + "/";
```
Comment (MaxKellermann) [inline src/storage/plugins/CurlStorage.cxx]: `.push_back('/')` is cheaper.

File: src/storage/plugins/CurlStorage.cxx [hunk 3]
```diff
@@ -77,9 +78,18 @@ CurlStorage::MapUTF8(const char *uri_utf8) const noexcept
...
+		path_esc = path_esc + elt_esc;
```
Comment (MaxKellermann) [inline src/storage/plugins/CurlStorage.cxx]: `+=` is cheaper.

**Comment by jwise** [comment]
I've made those changes.  Thanks for the style review!

**Comment by MaxKellermann** [comment]
29f78b18b1dd77afd517a519afe4acd23dd76407
cherry-picked to v0.20.x

**Pull request closed without merging**